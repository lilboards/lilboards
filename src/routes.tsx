import { Route } from 'react-router-dom';

import Layout from './components/Layout';
import Protected from './components/Protected';
import Board from './pages/Board';
import Boards from './pages/Boards';
import Home from './pages/Home';
import Login from './pages/Login';
import Logout from './pages/Logout';
import NotFound from './pages/NotFound';

const routes = (
  <Route path="/" element={<Layout />}>
    <Route index element={<Home />} />

    <Route path="/login" element={<Login />} />

    <Route path="/logout" element={<Logout />} />

    <Route
      path="/boards"
      element={
        <Protected check="email">
          <Boards />
        </Protected>
      }
    />

    <Route
      path="/boards/:boardId"
      element={
        <Protected check="id" signInAnonymously>
          <Board />
        </Protected>
      }
    />

    <Route path="*" element={<NotFound />} />
  </Route>
);

export default routes;
