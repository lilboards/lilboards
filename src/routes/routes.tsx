import { Route } from 'react-router-dom';
import Layout from 'src/components/Layout';
import Protected from 'src/components/Protected';
import Board from 'src/pages/Board';
import Boards from 'src/pages/Boards';
import ErrorBoundary from 'src/pages/ErrorBoundary';
import Home from 'src/pages/Home';
import List from 'src/pages/List';
import Lists from 'src/pages/Lists';
import Login from 'src/pages/Login';
import Logout from 'src/pages/Logout';
import NotFound from 'src/pages/NotFound';
import Support from 'src/pages/Support';

const routes = (
  <Route path="/" element={<Layout />}>
    <Route errorElement={<ErrorBoundary />}>
      <Route index element={<Home />} />

      <Route path="login" element={<Login />} />
      <Route path="logout" element={<Logout />} />
      <Route path="support" element={<Support />} />

      <Route path="boards">
        <Route element={<Protected check="email" />}>
          <Route index element={<Boards />} />
        </Route>

        <Route element={<Protected check="id" signInAnonymously />}>
          <Route path=":boardId" element={<Board />} />
        </Route>
      </Route>

      <Route path="lists">
        <Route element={<Protected check="email" />}>
          <Route index element={<Lists />} />
        </Route>

        <Route element={<Protected check="id" signInAnonymously />}>
          <Route path=":listId" element={<List />} />
        </Route>
      </Route>

      <Route path="*" element={<NotFound />} />
    </Route>
  </Route>
);

export default routes;
