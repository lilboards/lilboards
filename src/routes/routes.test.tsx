import routes from './routes';

it('matches snapshot', () => {
  expect(routes).toMatchSnapshot();
});
