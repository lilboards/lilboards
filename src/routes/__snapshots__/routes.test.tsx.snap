// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Route
  element={<Layout />}
  path="/"
>
  <Route
    errorElement={<ErrorBoundary />}
  >
    <Route
      element={<HomeLoader />}
      index={true}
    />
    <Route
      element={<LoginLoader />}
      path="login"
    />
    <Route
      element={<LogoutLoader />}
      path="logout"
    />
    <Route
      element={<SupportLoader />}
      path="support"
    />
    <Route
      path="boards"
    >
      <Route
        element={
          <ProtectedLoader
            check="email"
          />
        }
      >
        <Route
          element={<BoardsLoader />}
          index={true}
        />
      </Route>
      <Route
        element={
          <ProtectedLoader
            check="id"
            signInAnonymously={true}
          />
        }
      >
        <Route
          element={<BoardLoader />}
          path=":boardId"
        />
      </Route>
    </Route>
    <Route
      path="lists"
    >
      <Route
        element={
          <ProtectedLoader
            check="email"
          />
        }
      >
        <Route
          element={<ListsLoader />}
          index={true}
        />
      </Route>
      <Route
        element={
          <ProtectedLoader
            check="id"
            signInAnonymously={true}
          />
        }
      >
        <Route
          element={<ListLoader />}
          path=":listId"
        />
      </Route>
    </Route>
    <Route
      element={<NotFoundLoader />}
      path="*"
    />
  </Route>
</Route>
`;
